cpp/src/Ice/EndpointFactoryManager.cpp

```diff
@@ -49,8 +48,7 @@ IceInternal::EndpointFactoryManager::add(const EndpointFactoryPtr& factory)
 EndpointFactoryPtr
 IceInternal::EndpointFactoryManager::get(int16_t type) const
 {
-    lock_guard lock(_mutex); // TODO: Necessary?
-
+    lock_guard lock(_mutex);
```

Comment:
Previously the lock was required because "destroy" clears the map, and "get" can be iterating it concurrently.

Now with the removal of destroy, I think it is still required in case plug-ins load endpoint-factories, while a remote call access the factories from a separate thread. Not very likely but seems safer to keep the lock.

Reply:
We should clarify if plugin initialization can occur concurrently while other threads are using/calling the communicator.

In my view, the answer should be No. I don't see a use-case for allowing such concurrent initialization.